eclass/ecm-common.eclass

```diff
@@ -0,0 +1,236 @@
+	[[ ${ECM_INSTALL_FILES@a} == *a* ]] ||
+		die "ECM_INSTALL_FILES must be an array"
+fi
+
+# @FUNCTION: _ecm-common_preamble
+# @INTERNAL
+# @DESCRIPTION:
+# Create a CMakeLists.txt file with minimum ECM setup.
+_ecm-common_preamble() {
+	cat > CMakeLists.txt <<- _EOF_ || die
+	cmake_minimum_required(VERSION 3.16)
+	project(${PN} VERSION ${PV})
+
+	find_package(ECM "5.115.0" REQUIRED NO_MODULE)
+	set(CMAKE_MODULE_PATH \${CMAKE_CURRENT_SOURCE_DIR}/cmake/modules \${ECM_MODULE_PATH})
+
+	set(KDE_INSTALL_DOCBUNDLEDIR "${EPREFIX}/usr/share/help" CACHE PATH "")
+
+	include(KDEInstallDirs)
+	include(ECMOptionalAddSubdirectory) # commonly used
+	include(FeatureSummary)
+
+	_EOF_
+
+	if [[ ${ECM_INSTALL_ICONS} ]]; then
+		cat >> CMakeLists.txt <<- _EOF_ || die
+		include(ECMInstallIcons)
+
+		_EOF_
+	fi
+}
+
+# @FUNCTION: _ecm-common_i18n
+# @INTERNAL
+# @DESCRIPTION:
+# Find KF[56]I18n and call ki18n_install(po).
+_ecm-common_i18n() {
+	cat >> CMakeLists.txt <<- _EOF_ || die
+	find_package(KF6I18n)
+	if(NOT KF6I18n_FOUND)
+		find_package(KF5I18n REQUIRED)
+	endif()
+	ki18n_install(po)
+
+	_EOF_
+}
+
+# @FUNCTION: _ecm-common_docs
+# @INTERNAL
+# @DESCRIPTION:
+# Find KF[56]DocTools, call kdoctools_install(po) and
+# add_subdirectory(${ECM_HANDBOOK_DIR})
+_ecm-common_docs() {
+	cat >> CMakeLists.txt <<- _EOF_ || die
+	find_package(KF6DocTools)
+	if(NOT KF6DocTools_FOUND)
+		find_package(KF5DocTools REQUIRED)
+	endif()
+	kdoctools_install(po)
+
+	_EOF_
+
+	if [[ -d ${ECM_HANDBOOK_DIR} ]]; then
+		cat >> CMakeLists.txt <<- _EOF_ || die
+		add_subdirectory(${ECM_HANDBOOK_DIR})
+
+		_EOF_
+	fi
+}
+
+# @FUNCTION: _ecm-common_ecm_install_icons
+# @INTERNAL
+# @DESCRIPTION:
+# Installs icons listed in ECM_INSTALL_ICONS using ecm_install_icons
+_ecm-common_ecm_install_icons() {
+	local i
+	for i in "${ECM_INSTALL_ICONS[@]}"; do
+		cat >> CMakeLists.txt <<- _EOF_ || die
+		ecm_install_icons(ICONS ${i%:*} DESTINATION ${i#*:})
+		_EOF_
+	done
+}
+
+
+# @FUNCTION: _ecm-common_ecm_install_files
+# @INTERNAL
+# @DESCRIPTION:
+# Installs files listed in ECM_INSTALL_FILES using install(FILES ...)
+_ecm-common_ecm_install_files() {
+	local i
+	for i in "${ECM_INSTALL_FILES[@]}"; do
+		cat >> CMakeLists.txt <<- _EOF_ || die
+		install(FILES ${i%:*} DESTINATION ${i#*:})
+		_EOF_
+	done
+}
+
+# @FUNCTION: ecm-common_inject_heredoc
+# @DESCRIPTION:
+# Override this to inject custom Heredoc into the root CMakeLists.txt
+ecm-common_inject_heredoc() {
+	debug-print-function ${FUNCNAME} "$@"
+}
+
+# @FUNCTION: _ecm-common_summary
+# @INTERNAL
+# @DESCRIPTION:
+# Just calls feature_summary
+_ecm-common_summary() {
+	cat >> CMakeLists.txt <<- _EOF_ || die
+
+	feature_summary(WHAT ALL INCLUDE_QUIET_PACKAGES FATAL_ON_MISSING_REQUIRED_PACKAGES)
+	_EOF_
+}
+
+# @FUNCTION: ecm-common_src_prepare
+# @DESCRIPTION:
+# Wrapper for cmake_src_prepare with a Heredoc replacing the standard
+# root CMakeLists.txt file to only generate and install translations.
+ecm-common_src_prepare() {
```

Comment:
I find the whitespace between lines ugly here but you're the one doing most of the work, so wfm

Reply:
me too.